Backend/lib/beckn-services/src/AWS/S3/Flow.hs

```diff
@@ -243,3 +247,44 @@ mockDelete baseDirectory bucketName path =
 
 getFullPathMock :: String -> Text -> String -> String
 getFullPathMock baseDirectory bucketName path = baseDirectory <> "/" <> cs bucketName <> "/" <> path
+
+-- | Generate a pre-signed URL for uploading a video
+generateUploadUrl' ::
+  ( MonadFlow m
+  ) =>
+  Text ->
+  String ->
+  Seconds ->
+  m Text
+generateUploadUrl' = generateUrl' PUT
+
+-- | Generate a pre-signed URL for downloading a video
+generateDownloadUrl' ::
+  ( MonadFlow m
+  ) =>
+  Text ->
+  String ->
+  Seconds ->
+  m Text
+generateDownloadUrl' = generateUrl' GET
+
+data Method = GET | PUT
+
+generateUrl' ::
+  ( MonadFlow m
+  ) =>
+  Method ->
+  Text ->
+  String ->
+  Seconds ->
+  m Text
+generateUrl' method bucketName path expires = withLogTag "S3" $ do
+  env <- Amazonka.newEnv Amazonka.discover
+  now <- getCurrentTime
+  let bucketName' = Amazonka.BucketName bucketName
+      path' = Amazonka.ObjectKey $ T.pack path
+      expires' = fromInteger $ toInteger expires
+  bsUrl <- case method of
+    GET -> Amazonka.presignURL env now expires' $ Amazonka.newGetObject bucketName' path'
+    PUT -> Amazonka.presignURL env now expires' $ Amazonka.newPutObject bucketName' path' ""
```

Comment:
@coderabbitai is there a way to restrict file size and extension when user upload via presigned link?